erigon-lib/downloader/downloader.go

```diff
@@ -339,54 +340,52 @@ func (d *Downloader) ReCalcStats(interval time.Duration) {
 			}
 			stats.BytesCompleted += uint64(t.BytesCompleted())
 			stats.BytesTotal += uint64(t.Length())
-			if t.Complete.Bool() {
-				break //of switch
-			}
 
 			progress := float32(float64(100) * (float64(t.BytesCompleted()) / float64(t.Length())))
 			if progress == 0 {
 				zeroProgress = append(zeroProgress, t.Name())
-				break //of switch
 			}
 
 			d.logger.Log(d.verbosity, "[snapshots] progress", "file", t.Name(), "progress", fmt.Sprintf("%.2f%%", progress), "peers", len(peersOfThisFile), "webseeds", len(weebseedPeersOfThisFile))
 			isDiagEnabled := diagnostics.TypeOf(diagnostics.SegmentDownloadStatistics{}).Enabled()
 			if d.verbosity < log.LvlInfo || isDiagEnabled {
-
-				// more detailed statistic: download rate of each peer (for each file)
 				websRates := uint64(0)
-				webseedRates := make([]interface{}, 0, len(weebseedPeersOfThisFile)*2)
-				for _, peer := range weebseedPeersOfThisFile {
-					urlS := strings.Trim(strings.TrimPrefix(peer.String(), "webseed peer for "), "\"")
-					if urlObj, err := url.Parse(urlS); err == nil {
-						if shortUrl, err := url.JoinPath(urlObj.Host, urlObj.Path); err == nil {
-							dr := uint64(peer.DownloadRate())
-							webseedRates = append(webseedRates, shortUrl, fmt.Sprintf("%s/s", common.ByteCount(dr)))
-							websRates += dr
+				peersRates := uint64(0)
+				// more detailed statistic: download rate of each peer (for each file)
+				if !t.Complete.Bool() && progress != 0 {
+					webseedRates := make([]interface{}, 0, len(weebseedPeersOfThisFile)*2)
```

Comment:
`:=` creating variable invisible from outside of scope

Reply:
updated